src/users/users.service.ts

```diff
@@ -1,26 +1,46 @@
 import { Injectable } from '@nestjs/common';
 import { CreateUserDto } from './dto/create-user.dto';
 import { UpdateUserDto } from './dto/update-user.dto';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { NotFoundException } from '@nestjs/common';
 
 @Injectable()
 export class UsersService {
-  create(createUserDto: CreateUserDto) {
-    return 'This action adds a new user';
+  constructor(private readonly prisma: PrismaService) {}
+
+  async create(createUserDto: CreateUserDto) {
+      return await this.prisma.user.create({
+        data: createUserDto,
+      });
   }
 
   findAll() {
     return `This action returns all users`;
   }
 
-  findOne(id: number) {
-    return `This action returns a #${id} user`;
+  async findOne(id: string) {
+    const user = await this.prisma.user.findUnique({
+      where: { id },
+    });
+    if (!user) {
+      throw new NotFoundException(`User with ID ${id} not found`);
```

Comment:
Como vamos usar o decorator que recupera dados do usuário do header da requisição, presumimos que a autenticação já foi realizada.
Se o ID não for encontrado, isso indica um problema mais fundamental, como um erro na autenticação ou uma falha na extração do ID do usuário.
Lançar uma exceção aqui pode fazer com que o processo de debug seja mais complexo.